src/hooks/search/use-search.ts

```diff
@@ -1,6 +1,24 @@
 import { useState, useCallback } from 'react'
 import { type SearchDialogResult, type SearchApiResponse } from './types'
 
+function getItemUrl(item: SearchDialogResult): string {
+  console.log('item', item.modules)
+  switch (item.entity) {
+    case 'forum':
+      return `/forum/${item.id}/${item.title
+        ?.toLowerCase()
+        .replace(/\s+/g, '-')}`
+    case 'user':
+      return `/user/${item.username}`
+    case 'classroom':
+      return `/mentorship/${item.modules?.[0].videos?.[0].id}`
```

Comment:
Please consider adding early validation checks at the beginning of this method (before the switch statement) to ensure `item.modules` exists and has at least one element. Without this, accessing `item.modules[0]` could lead to runtime errors (e.g., undefined access) if `modules` is missing or empty, which might cause unexpected behavior or crashes. For example, you could add a guard clause like `if (!item.modules?.length) return null;` to handle this safely.